lib/explorer/series.ex

```diff
@@ -4276,19 +4276,39 @@ defmodule Explorer.Series do
 
   ## Examples
 
-      iex> s = Explorer.Series.from_list(["  abc", "def  ", "  bcd"])
+      iex> s = Explorer.Series.from_list(["  abc", "def  ", "  bcd   "])
       iex> Explorer.Series.trim(s)
       #Explorer.Series<
         Polars[3]
         string ["abc", "def", "bcd"]
       >
+
+      iex> s = Explorer.Series.from_list(["abc", "adefa", "bcda"])
+      iex> Explorer.Series.trim(s, "a")
+      #Explorer.Series<
+        Polars[3]
+        string ["bc", "def", "bcd"]
+      >
+
+      iex> s = Explorer.Series.from_list(["£123", "£1.00", "£1.00£"])
+      iex> Explorer.Series.trim(s, "£")
+      #Explorer.Series<
+        Polars[3]
+        string ["123", "1.00", "1.00"]
+      >
   """
   @doc type: :string_wise
   @spec trim(Series.t()) :: Series.t()
   def trim(%Series{dtype: :string} = series),
-    do: apply_series(series, :trim)
+    do: apply_series(series, :trim, [""])
+
+  @doc type: :string_wise
+  @spec trim(Series.t(), String.t()) :: Series.t()
+  def trim(%Series{dtype: :string} = series, string),
```

Comment:
Here we could have a guard:

```suggestion
  def trim(%Series{dtype: :string} = series, string) when is_binary(string),
```